packages/identity/src/user-identity.ts

```diff
@@ -45,74 +38,27 @@ export class UserIdentity extends Construct {
     super(scope, id);
 
     // Unless explicitly stated, created a default Cognito User Pool and Web Client.
-    if (!props?.userPool) {
-      this.userPool = new UserPool(this, "UserPool", {
-        deletionProtection: true,
-        passwordPolicy: {
-          minLength: 8,
-          requireLowercase: true,
-          requireUppercase: true,
-          requireDigits: true,
-          requireSymbols: true,
-          tempPasswordValidity: Duration.days(3),
-        },
-        mfa: Mfa.REQUIRED,
-        accountRecovery: AccountRecovery.EMAIL_ONLY,
-        autoVerify: {
-          email: true,
-        },
-      });
-
-      (this.userPool.node.defaultChild as CfnUserPool).userPoolAddOns = {
-        advancedSecurityMode: "ENFORCED",
-      };
-
-      const stack = Stack.of(this);
-
-      ["AwsSolutions-IAM5", "AwsPrototyping-IAMNoWildcardPermissions"].forEach(
-        (RuleId) => {
-          PDKNag.addResourceSuppressionsByPathNoThrow(
-            stack,
-            `${PDKNag.getStackPrefix(stack)}${id}/UserPool/smsRole/Resource`,
-            [
-              {
-                id: RuleId,
-                reason:
-                  "MFA requires sending a text to a users phone number which cannot be known at deployment time.",
-                appliesTo: ["Resource::*"],
-              },
-            ]
-          );
-        }
-      );
+    this.userPool = !props?.userPool
+      ? new UserPoolWithMfa(this, "UserPool")
+      : props.userPool;
 
-      this.userPoolClient = this.userPool.addClient("WebClient", {
-        authFlows: {
-          userPassword: true,
-          userSrp: true,
-        },
-      });
-    } else {
-      this.userPool = props.userPool;
-    }
+    this.identityPool = new IdentityPool(
+      this,
+      "IdentityPool",
+      props?.identityPoolOptions
+    );
 
-    this.identityPool = new IdentityPool(this, "IdentityPool", {
-      ...props?.identityPoolOptions,
-      authenticationProviders: {
-        ...props?.identityPoolOptions?.authenticationProviders,
-        userPools: [
-          ...(props?.identityPoolOptions?.authenticationProviders?.userPools ||
-            []),
-          ...(!props?.userPool
-            ? [
-                new UserPoolAuthenticationProvider({
-                  userPool: this.userPool,
-                  userPoolClient: this.userPoolClient!,
-                }),
-              ]
-            : []),
-        ],
+    this.userPoolClient = this.userPool.addClient("WebClient", {
```

Comment:
What if user pool is supplied in options and already contains "WebClient" client? Does UserPool support duplicate client names? Can we check if "WebClient" client already exists, and either throw an error or use the existing client?

Reply:
Updated.